### PR TITLE
ports wendigo from snowmap

### PR DESCRIPTION
**Pull request by MadmanMartian**
Changes I've made to this is
- Removed the redundant 'global list of wendigo', in favour of using animal_count, amongst other removed redundant code
- Speak name returns "unknown" if there is no names in its mimic list, for spookiness.
- Attempted to recreate whatever speed enforcement they had previously, as they used 'speed' variables, but all would go sanik fast
- Added an exception in their CanAttack so they don't focus down on dead mobs forever

### CODE REVIEW

**Comment by FudgePucker** [comment]
Wait I thought that the wendingo was the skifree monster, did they change it?

**Comment by MadmanMartian** [comment]
skifree is a subtype of wendigo, it's at the bottom of wendigo.dm. I added its sprite amongst other snowmap mobs in the deer PR

File: code/modules/mob/living/simple_animal/hostile/wendigo.dm [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	attack_sound = 'sound/weapons/heavysmash.ogg'
+	environment_smash_flags = SMASH_LIGHT_STRUCTURES | SMASH_CONTAINERS | SMASH_WALLS
+	var/enraged = 0
+	var/punch_throw_chance = 25
+	var/punch_throw_speed = 1
+	var/punch_throw_range = 10
+
+/mob/living/simple_animal/hostile/wendigo/alpha/Life()
+	..()
+	if(health < (maxHealth/2) && enraged == 0)
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/hostile/wendigo.dm]: do you need a `isDead()` check before this?

File: code/modules/reagents/reagent_containers/food/snacks/meat.dm [hunk 2]
```diff
@@ -235,3 +235,18 @@ var/global/list/valid_random_food_types = existing_typesof(/obj/item/weapon/reag
...
 	reagents.add_reagent(PETRITRICIN, 3)
 
+/obj/item/weapon/reagent_containers/food/snacks/meat/wendigo
+	name = "strange meat"
+	desc = "Doesn't look very appetizing, but if you're considerably hungry..."
+	icon_state = "wendigo_meat"
+
+/obj/item/weapon/reagent_containers/food/snacks/meat/wendigo/New()
+	..()
+	reagents.add_reagent("nutriment", rand(10,25))
```
Comment (DamianX) [inline code/modules/reagents/reagent_containers/food/snacks/meat.dm]: use `NUTRIMENT`

File: code/modules/reagents/reagent_containers/food/snacks/meat.dm [hunk 3]
```diff
@@ -235,3 +235,18 @@ var/global/list/valid_random_food_types = existing_typesof(/obj/item/weapon/reag
...
+	src.bitesize = 30
```
Comment (DamianX) [inline code/modules/reagents/reagent_containers/food/snacks/meat.dm]: put this in the declaration

File: code/modules/reagents/reagent_containers/food/snacks/meat.dm [hunk 4]
```diff
@@ -235,3 +235,18 @@ var/global/list/valid_random_food_types = existing_typesof(/obj/item/weapon/reag
...
+
+/obj/item/weapon/reagent_containers/food/snacks/meat/wendigo/consume(mob/living/carbon/eater, messages = 0)
+	..()
```
Comment (DamianX) [inline code/modules/reagents/reagent_containers/food/snacks/meat.dm]: you should return the parent's return value

File: code/modules/mob/living/simple_animal/hostile/wendigo.dm [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+	Let them evolve and heal from eating. Should a carbon they eat go over a certain damage threshold and
+	the wendigo is at full health. Strip and delete the corpse.
+	Human wendigo->1 eat->Wendigo->3 eat->Alpha Wendigo (Can only be ONE at any time)
+	They don't like fire. Will stay away from fire (As far as their light radius -1) and won't target anyone
+	in vicinity of said fire
+	Can mimic voices, steal poly code, and change it (Maybe just keep a log of people consumed,
+	and use their name plus shout for help)
+	Butcherable for one piece of meat only. When consumed (Don't use reagents) will begin to turn the person
+*/
+#define HUMEVOLV 1
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/hostile/wendigo.dm:11]: this is gonna sound petty but it's not entirely clear what these stand for

File: code/modules/mob/living/simple_animal/hostile/wendigo.dm [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+			else
+				target_turf = get_ranged_target_turf(T, dir, punch_throw_range)
+			M.throw_at(target_turf,100,punch_throw_speed)
+
+/mob/living/simple_animal/hostile/wendigo/skifree
+	name = "yeti"
+	desc = "holding F might not work this time"
+	icon_state = "yeti"
+	icon_living = "yeti"
+	speed = -1
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/hostile/wendigo.dm]: image

File: code/modules/mob/living/simple_animal/hostile/wendigo.dm [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+			var/turf/target_turf
+			if(istype(T, /turf/space)) // if ended in space, then range is unlimited
+				target_turf = get_edge_target_turf(T, dir)
+			else
+				target_turf = get_ranged_target_turf(T, dir, punch_throw_range)
+			M.throw_at(target_turf,100,punch_throw_speed)
+
+/mob/living/simple_animal/hostile/wendigo/skifree
+	name = "yeti"
+	desc = "holding F might not work this time"
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/hostile/wendigo.dm]: capitalization, punctuation and ??? I guess this is a reference?

File: code/modules/mob/living/simple_animal/hostile/wendigo.dm [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+		var/dist = get_dist(the_target, fire)
+		if(dist < (fire.light_range*2))//Just sitting on the edge of the fire
+			return
+
+	if(isliving(target) && !ishuman(target))
+		var/mob/living/L = target
+		if(L.isDead())
+			return
+
+	return ..(the_target)
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/hostile/wendigo.dm]: no need to explicitly pass `the_target`

File: code/modules/mob/living/simple_animal/hostile/wendigo.dm [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+					if(mob_target.health < -400)
+						visible_message("<span class = 'warning'>\The [src] is trying to consume \the [mob_target]!</span>","<span class = 'warning'>You hear crunching.</span>")
+						if(do_after(src, mob_target, 50, needhand = FALSE))
+							consumes += 1
+							if(!names.Find(mob_target.real_name))
+								names.Add(mob_target.real_name)
+							mob_target.gib()
+
+			return
+	. =..()
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/hostile/wendigo.dm]: I don't think you need this but correct me if I'm wrong, don't just believe me

File: code/modules/mob/living/simple_animal/hostile/wendigo.dm [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+				mob_target.adjustBruteLoss(damage)
+				if(health < maxHealth)
+					health = min(maxHealth,(health+damage))
+
+				if(ishuman(mob_target))
+					if(mob_target.health < -400)
+						visible_message("<span class = 'warning'>\The [src] is trying to consume \the [mob_target]!</span>","<span class = 'warning'>You hear crunching.</span>")
+						if(do_after(src, mob_target, 50, needhand = FALSE))
+							consumes += 1
+							if(!names.Find(mob_target.real_name))
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/hostile/wendigo.dm]: replace this with `names |= mob_target.real_name`

File: code/modules/mob/living/simple_animal/hostile/wendigo.dm [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
+			return
+	. =..()
+
+/mob/living/simple_animal/hostile/wendigo/Life()
+	/*Check evolve like zombies, run away from fire,
+	mimic speech if anyone's nearby but just out of eyesight
+	*/
+	..()
+	if(!isDead())
+		check_evolve()
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/hostile/wendigo.dm]: this proc may `qdel(src)`, surely we don't want to do anything else in here if that's the case
I suggest you make it return a value and check it

**Review by DamianX — DISMISSED** [review]
\>mapping (snowmap)
\>implying

File: code/modules/mob/living/simple_animal/hostile/wendigo.dm (hunk 7 above)
Comment (Kurfursten) [reply code/modules/mob/living/simple_animal/hostile/wendigo.dm]: You can hold F to go faster than the Skifree yeti

File: code/modules/mob/living/simple_animal/hostile/wendigo.dm (hunk 9 above)
Comment (unid15) [reply code/modules/mob/living/simple_animal/hostile/wendigo.dm]: could be replaced with ``return ..()`` but really there's no issue here

File: code/modules/mob/living/simple_animal/hostile/wendigo.dm (hunk 5 above)
Comment (MadmanMartian) [reply code/modules/mob/living/simple_animal/hostile/wendigo.dm:11]: far as I could tell, they're defines for the specific wendigo evolutions. HUMEVOLV is for the human looking one, and the other's for the weird and silly looking one.

Big one doesn't evolve, so no define for it.

**Comment by jknpj** [comment]
These things are busted as fuck but i like them.

File: code/datums/diseases/wendigo_transformation.dm [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+			if(5)
+				if(prob(50))
+					to_chat(H, "<span class = 'warning'>Our mind hurts.</span>")
+					H.adjustBrainLoss(25)
+					H.hallucination += 20
+				if(prob(15))
+					var/mob/living/simple_animal/hostile/wendigo/human/W = new/mob/living/simple_animal/hostile/wendigo/human(H.loc)
+					W.names += H.real_name
+					H.drop_all()
+					qdel(H)
```
Comment (jknpj) [inline code/datums/diseases/wendigo_transformation.dm:43]: Why not gib them? Way more dramatic. I know, it's petty.

Comment (MadmanMartian) [reply code/datums/diseases/wendigo_transformation.dm:43]: that does sound like it could be fun.

**Pull request merged**